addons/gamification/models/goal.py

```diff
@@ -221,7 +221,7 @@ def _check_remind_delay(self):
 
         # generate a reminder report
         body_html = self.env.ref('gamification.email_template_goal_reminder')._render_field('body_html', self.ids, compute_lang=True)[self.id]
-        self.message_notify(
+        self.env['mail.thread'].message_notify(
```

Comment:
You can give model and res_id to link the message to a record (notably for acls). It should support non thread models, so let us try to give those two and see if it works ?